site/content/docs/multistep-checks/example-checks.md

```diff
@@ -0,0 +1,196 @@
+
+An API_KEY from https://crudapi.co.uk/ is required for this example to work.
+
+```ts
+import { test, expect } from "@playwright/test"
+
+const baseUrl = "https://crudapi.co.uk/api/v1/todo"
+
+const headers = {
+  Authorization: `Bearer ${process.env.CRUD_API_KEY}`,
+  "Content-Type": "application/json",
+}
+
+/**
+ * Share state between hooks and test.steps
+ */
+let createdResources = null
+
+/**
+ * Use `beforeAll` as a setup script
+ * to make sure no todo's exist.
+ */
+test.beforeAll(async ({ request }) => {
+  const response = await request.get(baseUrl, { headers })
+  const { items } = await response.json()
+  expect(items.length).toEqual(0)
+})
+
+/**
+ * Use `afterAll` as a teardown script
+ * to remove the created todo before the check ends
+ */
+test.afterAll(async ({ request }) => {
+  if (createdResources) {
+    const response = await request.delete(baseUrl, {
+      headers,
+      data: createdResources,
+    })
+    expect(response).toBeOK()
+  }
+})
+
+test("Todo List", async ({ request }) => {
+  await test.step("Create a Todo", async () => {
+    const response = await request.post(baseUrl, {
+      headers,
+      data: [
+        {
+          title: "Testing Checkly Multistep checks",
+          done: true,
+        },
+      ],
+    })
+
+    expect(response).toBeOK()
+    const { items } = await response.json()
+    createdResources = items
+  })
+})
+
+```
+
+### Verifying binary payloads
+One way to ensure that e.g CDN endpoints are serving images correctly is by verifying its binary. 
+
+Here is an example of how to verify several images in a loop, checking each image against a unique binary.
+
+```ts
+import { test, expect } from "@playwright/test"
+import crypto from "crypto"
+
+test("Verify Binary Payloads", async ({ request }) => {
+  // SpaceX Launch Images
+  const IMAGES_TO_VERIFY = [
+    {
+      url: "https://farm1.staticflickr.com/856/28684550147_49802752b3_o.jpg",
+      hash: "46df689c0016e4f06746f07b83546d5e",
+    },
+    {
+      url: "https://farm1.staticflickr.com/927/28684552447_956a9744f1_o.jpg",
+      hash: "ffb011da0c7cc45413c632ccd62947cf",
+    },
+    {
+      url: "https://farm2.staticflickr.com/1828/29700007298_8ac5891d2c_o.jpg",
+      hash: "eab74946120df579967922794e387276",
+    },
+    {
+      url: "https://farm1.staticflickr.com/914/29700004918_31ed7b73ef_o.jpg",
+      hash: "5e20e98a63522a0829aa5ad0003e52c6",
+    },
+  ]
+
+  for (const [index, { url, hash }] of IMAGES_TO_VERIFY.entries()) {
+    await test.step(`Fetch image #${index}`, async () => {
+      const response = await request.get(url)
+
+      const body = await response.body()
+      expect(crypto.createHash("md5").update(body).digest("hex")).toBe(hash)
+    })
+  }
+})
+```
+
+If you have suggestions for additional check examples to add here, please do not hesitate to reach out to our [support](mailto:support@checklyhq.com), or join our [Slack community](https://www.checklyhq.com/slack/).
```

Comment:
@drakirnosslin suggestion: we could also ask people to submit a PR, since the repo is public.